docs/concepts/components/callbacks.md

```diff
@@ -2,27 +2,92 @@
 title: Callbacks
 description: ComponentLink and Callbacks
 ---
-The component "link" is the mechanism through which components are able to register callbacks and update themselves.
+
+The component "link" is the mechanism through which components are able to create callbacks and update themselves.
 
 ## ComponentLink API
 
+### send_message
+
+Sends a message to the component.
+Messages are handled by the `update` method which decides whether the component should re-render.
+
+### send_message_batch
+
+Sends multiple messages to the component at the same time.
+This is similar to `send_message` but if any of the messages cause the `update` method to return `true`,
+the component will re-render after all messages in the batch have been processed.
+
+If the given vector is empty, this function doesn't do anything.
+
 ### callback
 
-Registers a callback that will send a message to the component's update mechanism when it is executed. Under the hood, it will call `send_self` with the message that is returned by the provided closure. A `Fn(IN) -> COMP::Message` is provided and a `Callback<IN>` is returned.
+Create a callback that will send a message to the component when it is executed.
+Under the hood, it will call `send_message` with the message that is returned by the provided closure.
+
+This method has a version that accepts an `FnOnce` instead, `callback_once`.
```

Comment:
Should this say something like (I feel it's not 100% clear about the name of the name of the alternative method `callback_once`):
```suggestion
There is a different method (called `callback_once`) which accepts a `FnOnce` instead of a `Fn`.
``` 
?

Reply:
I agree that this improves the flow but can we drop the parentheses?

Reply:
Yup.

Reply:
(n.b. I have updated the original suggestion)